src/zzz_od/application/shiyu_defense/shiyu_defense_team_utils.py

```
@@ -258,21 +256,102 @@ def calc_teams(
+            resistance_list.append(DmgTypeEnum.UNKNOWN)
+
+        team_list.append(DefensePhaseTeamInfo(weakness_list[:2], resistance_list[:2]))
+        team_list[-1].team_idx = 0  # 标记待配队，searcher 会覆盖
+
+    # 只传需要配队的房间给搜索器，跳过已通关（team_idx == -1）的房间
+    scored_indices = [i for i, t in enumerate(team_list) if t.team_idx != -1]
+    scored_teams = [team_list[i] for i in scored_indices]
+    if len(scored_teams) > 0:
+        searcher = DefenseTeamSearcher(ctx, scored_teams)
+        search_result = searcher.search()
+        for orig_idx, result_team in zip(scored_indices, search_result):
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)